Dockerfile

```diff
@@ -1,16 +1,18 @@
-FROM golang:1.20-alpine3.18 as builder
+FROM --platform=$BUILDPLATFORM golang:1.20-alpine3.18 as builder
 WORKDIR $GOPATH/src/go.k6.io/k6
-COPY . .
+ARG TARGETOS TARGETARCH
 RUN apk --no-cache add git=~2
-RUN CGO_ENABLED=0 go install -a -trimpath -ldflags "-s -w -X go.k6.io/k6/lib/consts.VersionDetails=$(date -u +"%FT%T%z")/$(git describe --tags --always --long --dirty)"
+RUN --mount=target=.,readonly \
+    --mount=type=cache,target=/root/.cache/go-build \
+	CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH go build -a -trimpath -o /usr/bin/k6
```

Comment:
Additionalyl I think `-a` compeltely negates the fact that we are havging a cache of the builds as `-a` basically saays to ignore that and rebuild